lib/phoenix_ecto/sql/sandbox.ex

```diff
@@ -32,21 +48,101 @@ defmodule Phoenix.Ecto.SQL.Sandbox do
   """
 
   import Plug.Conn
+  alias Plug.Conn
+  alias Phoenix.Ecto.SQL.SandboxSupervisor
+
+  @doc """
+  Spawns a sandbox process to checkout a connection for a remote client.
+
+  ## Examples
+
+      iex> {:ok, _owner_pid, metdata} = checkout(MyApp.Repo)
+  """
+  def checkout(repo, opts \\ []) do
```

Comment:
I am not a big fan of the name because it can be mistaken by the sandbox checkin/checkout. Can we have something more obvious?

Reply:
`spawn_checkout` ?

Reply:
Given that this is the `SQL.Sandbox` module, we could also `spawn/1` and `shutdown/1` on the module? Makes sense with the module API and is distinct from checkout/checkin?

Reply:
Let's call it `start_link` and `stop` then? `{:ok, pid, term}` is a known start link return but we need to say the process is already started in a supervision tree so people are not tempted to start it in their supervision tree too.

Reply:
For the e2e use case, we won't want to link though since the caller will be exiting after starting the sandbox? Should be support both start and start_link?

Reply:
Gah, you are right. There is no link. it could be `start_child` and `stop` or `spawn` and `stop`. But I think spawn will be surprising in return type.